prez-lib/src/types.ts

```diff
@@ -1,183 +1,87 @@
-        };
-    },
-
-    prezLiteral(paramsOrString:{text: string, language?: string, dataType?: PrezNode}|string): PrezLiteral {
-        if(typeof(paramsOrString) == 'string') {
-            return {
-                termType: 'Literal',
-                text: paramsOrString
-            }
-        }
-        const {text, language, dataType} = paramsOrString;
-        return {
-            termType: 'Literal',
-            text,
-            language,
-            dataType
-        };
-    },
-
-    prezBlankNode(node: PrezProperties): PrezBlankNode {
-        return {
-          termType: 'BlankNode',
-          properties
-        };
-    }
-
+    links?: PrezLink[];
+    rdfTypes?: PrezNode[];
+    termType: "node";
 };
 
-
-
-
-
-export interface ListItem {
-    label?: string;
-    uri: string;
-    description?: string;
-    link?: string;
-    extras?: {
-        [key: string]: string;
-    };
+/**
+ * Represents a `prez:link` value, containing parent info
+ */
+export type PrezLink = {
+    value: string;
+    parents?: PrezNode[];
+    // maybe label?
 };
 
-export interface ObjectItem {
-    label?: string;
-    uri: string;
-    description?: string;
-    link?: string;
-    properties: {
-        predicate: string;
-        object: string;
-    }[];
+/**
+ * Represents an RDF Blank Node
+ */
+export type PrezBlankNode = PrezTerm & {
+    id: string;
+    properties: PrezProperty[];
+    termType: "blanknode";
 };
 
-// export interface ObjectRowProps {
-    
-// };
-
-// export interface ObjectTableProps {
-//     label?: string;
-//     uri: string;
-//     types: string[];
-//     baseClass: string;
-//     description?: string;
-//     geometries?: string[];
-//     properties: ObjectRowProps[];
-// };
-
-// export interface Value {
-//     value: string;
-//     qname?: string;
-//     label?: string;
-//     description?: string;
-//     provenance?: string;
-//     language?: string;
-// };
-
-export interface Prefixes {
-    [namespace: string]: string;
-};
-
-export interface Profile {
-    namespace: string;
-    token: string;
-    title: string;
-    description: string;
-    mediatypes: string[];
-    defaultMediatype: string;
-    labelPredicates: string[];
-    descriptionPredicates: string[];
-    explanationPredicates: string[];
+/**
+ * Represents a 'row' in an item table
+ */
+export type PrezProperty = {
+    predicate: PrezNode;
+    object: PrezTerm[];
 };
 
-export interface Mediatype {
-    title: string;
-    mediatype: string;
-    default: boolean;
+/**
+ * Represents a search result
+ */
+export type PrezSearchResult = {
+    hash: string;
+    weight: number;
+    predicate: PrezNode;
+    match: PrezLiteral;
+    resource: PrezItem;
 };
 
-export interface ProfileHeader {
-    default: boolean;
-    current: boolean;
-    uri: string;
-    token: string;
-    title: string;
-    description: string;
-    mediatypes: Mediatype[];
+// to be replaced by PrezItem?
```

Comment:
Does this mean PrezItem will replace ItemExtra?

Reply:
this was the plan yes unless there was something ItemExtra had that isn't covered